staging/src/github.com/kubeedge/mapper-framework/pkg/common/timer.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"time"
+)
+
+// Timer is to call a function periodically.
+type Timer struct {
+	Function func()
+	Duration time.Duration
+	Times    int
+}
+
+// Start start a timer.
+func (t *Timer) Start() {
```

Comment:
1. Add a `New()` function to create ticker;
1. Using the select{} to take channel value;
1. Input a cancelable context;
1. Ticker needs to be closed when the maximum number of times is reached;
1. Make `Start()` can only be call once;

Maybe a open source project is a good option, e.g., https://github.com/go-co-op/gocron